graphql_compiler/tests/schema_transformation_tests/test_split_query.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+import unittest
+
+from graphql import print_ast
+
+
+# The below namedtuple is used to check the structure of SubQueryNodes in tests
+ExampleQueryNode = namedtuple(
+    'ExampleQueryNode', (
+        'query_str',
+        'schema_id',
+        'child_query_nodes_and_out_names',
+        # List[Tuple[ExampleQueryNode, str, str]]
+        # child example query node, parent out name, child out name
+    )
+)
+
+
+class TestSplitQuery(unittest.TestCase):
+    def _check_query_node_structure(self, root_query_node, root_example_query_node):
+        """Check root_query_node has no parent and has the same structure as the example input."""
+        self.assertIs(root_query_node.parent_query_connection, None)
+        self._check_query_node_structure_helper(root_query_node, root_example_query_node)
+
+    def _check_query_node_structure_helper(self, query_node, example_query_node):
+        """Check query_node has the same structure as example_query_node."""
+        # Check AST and id of the parent
+        self.assertEqual(print_ast(query_node.query_ast), example_query_node.query_str)
+        self.assertEqual(query_node.schema_id, example_query_node.schema_id)
+        # Check number of children matches
+        self.assertEqual(len(query_node.child_query_connections),
+                         len(example_query_node.child_query_nodes_and_out_names))
+        for i in range(len(query_node.child_query_connections)):
```

Comment:
Consider instead doing something like `for i, (child_query_connection, expected_data) in enumerate(six.moves.zip(query_node.child_query_connections, example_query_node.child_query_nodes_and_out_names)):`

Then you get to avoid much of the `[i]` and destructuring below. Also consider assigning the values that get passed to `six.moves.zip` to some local variables, just to save on some typing and line complexity.